### PR TITLE
MABL-4454: support Java 11 runtime

### PR DESCRIPTION
**Pull request by bertold**
JAXB and Jackson libraries have been replaced with XStream (for XML serialization - no explicit dependency needed here as Jenkins itself is using this library) and Google's GSON library for JSON serialization. The new code will work with both Java 8 and Java 11 runtime.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java [hunk 1]
```diff
@@ -70,15 +71,7 @@
  */
 public class MablRestApiClientImpl implements MablRestApiClient {
 
-    private final ObjectMapper objectMapper = new ObjectMapper();
-
-    {
-        objectMapper.configure(FAIL_ON_UNKNOWN_PROPERTIES, false);
-        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```
Comment (twistedpair) [inline src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java:77]: I assume we get the same functionality by default from `gson`?

File: src/test/java/com/mabl/integration/jenkins/MablStepDeploymentRunnerTest.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public void setup() {
         );
     }
 
-    @Test
+    @Test(timeout = 6000000)
```
Comment (twistedpair) [inline src/test/java/com/mabl/integration/jenkins/MablStepDeploymentRunnerTest.java:82]: Should we really give this test 1 2/3 hours to run?

**Review by twistedpair — APPROVED** [review]
Nice work finding a way around JAXB and Java 8/11.

File: src/test/java/com/mabl/integration/jenkins/MablStepDeploymentRunnerTest.java (hunk 2 above)
Comment (bertold) [reply src/test/java/com/mabl/integration/jenkins/MablStepDeploymentRunnerTest.java:82]: Ooops! That's a mistake.

File: src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java (hunk 1 above)
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java:77]: Yes, we do. I am writing additional unit tests to confirm.

**Pull request merged**